lib/postgrex/protocol.ex

```diff
@@ -228,7 +236,7 @@ defmodule Postgrex.Protocol do
       :ok ->
         auth_recv(s, opts, <<>>)
       {:error, reason} ->
-        error(%Postgrex.Error{message: "tcp send: #{reason}"}, s)
+        error(Postgrex.Error.exception(:tcp, "send", reason), s)
```

Comment:
This and the remaining in this file error can be an `:ssl` error as well. Consider returning `{:error, exception}` from `msg_send` and `msg_recv`.